incentivization.md

```diff
@@ -0,0 +1,342 @@
+A payment method should generally have the following properties:
+- wide distribution;
+- good liquidity;
+- low latency;
+- good privacy;
+- high security.
+
+Let's list all decentralized payment options:
+- ETH;
+- a token on Ethereum (ERC20);
+- a token on another EVM-based blockchain or a rollup;
+- a token on a non-EVM blockchain (such as BTC / Lightning).
+
+Note also that there may be different market models that may motivate the choice of the payment method.
+One model assumes that each client pays for its own requests.
+Another model includes (centralized?) entities (i.e., developers of Waku-based apps) that pay for their users in bulk.
+
+We also note that:
+- eventually the protocol may support multiple payment methods;
+- however, the MVP version should be simple, which likely means supporting just one payment method;
+- if the initially supported payment method is an ERC-20 token, it should be easy to add other ERC-20 tokens later, including a potential WAKU token.
+
+# Evaluation
+
+We should think about what the success metrics for an incentivized protocol are, and how to measure them both in simulated settings, as well as in a live network.
+
+# Future work
+
+Let us now outline some of the open questions beyond MVP.
+## Price discovery
+
+To offer a reasonable price, a server should understand its costs.
+The costs of a Store server are storage, bandwidth, and computation.
+A Store server does two things: it stores messages, and serves messages to clients.
+
+The cost of storing messages is composed of:
+- storage:
+	- storage costs of all older messages: proportional to cumulative (message size x time stored);
+	- the cost for I/O operations for storing new messages (roughly constant per unit time, though may fluctuate due to caching, disk fragmentation, etc.);
+- bandwidth (download) for receiving new messages;
+- computational costs.
+
+The cost of serving messages to clients, per unit time, is composed of:
+- bandwidth
+	- upload: proportional to (number of clients) x (length of time frame requested) x (message size);
+	- download: proportional to the number of requests;
+- computational cost of handling requests.
+
+Storage is likely the dominating cost.
+Storage costs is proportional to the amount of information stored and the time it is stored for.
+A cumulative cost of storing a single message grows linearly with time.
+The number of messages in a response may be approximated by the length of the time frame requested.
+
+Computation: the server spends computing cycles while handling requests.
+This costs likely depends not only on the computation itself, but also at the database structure.
+For example, retrieving old or rarely requested messages from the local database may be more expensive than fresh or popular ones due to caching.
+
+More formal calculations should be done, under certain assumptions about message flow (i.e., that it is constant).
+
+## Price negotiation
```

Comment:
> this becomes unnecessary if we can introduce some kind of market mechanism

These are not mutually exclusive. I see the "formal calculations" not as something that the whole network would agree on, but rather a suggested formula that each server would run locally, using its own costs of storage, bandwidth, etc. It would then advertise the calculated price (plus margin) and compete in the marketplace.

> a dedicated topic (or other mechanism) for store nodes to advertise their services and costs

For comparison: in Lightning, nodes advertise their channel fees in the P2P network. They can't do it too often, and a payment may fail if the price has just been updated but the sender is yet unaware of this, but overall this mechanism works OK in practice.

Reply:
Captured the ideas in #51 and #52, resolving this thread.